addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,575 @@
+
+        # NOTE: TicketBai supports simplified invoices WITH recipients: use checkbox in invoice (optional until POS implemented)
+        # NOTE: credit notes for simplified invoices are ALWAYS simplified BUT can have a recipient even if invoice doesn't
+        if invoice._is_l10n_es_tbai_simplified():
+            return values  # do not set 'recipient' unless there is an actual recipient (used as condition in template)
+
+        # === RECIPIENTS (DESTINATARIOS) ===
+        eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
+
+        nif = False
+        alt_id_country = False
+        partner = invoice.commercial_partner_id
+        alt_id_number = partner.vat or 'NO_DISPONIBLE'
+        alt_id_type = ""
+        if (not partner.country_id or partner.country_id.code == 'ES') and partner.vat:
+            # ES partner with VAT.
+            nif = partner.vat[2:] if partner.vat.startswith('ES') else partner.vat
+        elif partner.country_id.code in eu_country_codes:
+            # European partner
+            alt_id_type = '02'
+        else:
+            # Non-european partner
+            if partner.vat:
+                alt_id_type = '04'
+            else:
+                alt_id_type = '06'
+            if partner.country_id:
+                alt_id_country = partner.country_id.code
+
+        values_dest = {
+            'nif': nif,
+            'alt_id_country': alt_id_country,
+            'alt_id_number': alt_id_number,
+            'alt_id_type': alt_id_type,
+            'partner': partner,
+            'partner_address': ', '.join(filter(lambda x: x, [partner.street, partner.street2, partner.city])),
+        }
+
+        values.update({
+            'recipient': values_dest,
+            # NOTE: for Bizkaia, option below would be "D" for purchase invoices (not supported yet)
+            # third party = Tercero, recipient = Destinatario, sender = No (not thirdparty or recipient)
+            # NOTE: option T could be useful for companies handling invoicing for third parties
+            'thirdparty_or_recipient': "N",
+        })
+        return values
+
+    def _l10n_es_tbai_get_invoice_values(self, invoice, cancel):
+        # Header
+        values = {'invoice': invoice}
+        if cancel:
+            return values
+
+        # Credit notes (factura rectificativa)
+        # NOTE values below would have to be adapted for purchase invoices (Bizkaia LROE)
+        values['is_refund'] = invoice.move_type == 'out_refund'
+        if values['is_refund']:
+            values['credit_note_code'] = invoice.l10n_es_tbai_refund_reason
+            values['credit_note_invoice'] = invoice.reversed_entry_id
+
+        # Lines (detalle)
+        values['invoice_lines'] = [{
+            'line': line,
+            'discount': (discount := line.price_subtotal / (1 - line.discount / 100) * line.discount / 100),
```

Comment:
@odoostan What if the invoice is in foreign currency?  The values from SII are all in EUR or company currency and this should be the case here as well.  So, we should use line.balance instead...

Reply:
Has anyone considered clearer names for currency fields in local and foreign currency ?
I guess this applies to the line below as well, for computing `unit_price`